### PR TITLE
fix: avoid bfcache optimisation for page unloaded events

### PR DESCRIPTION
**Pull request by saikumarrs**
## PR Description

I've fixed the issue with page unloaded events firing functionality by forcefully subscribing to "beforeunloaded" event when this feature is enabled.
However, we need to compromise on the bfcache optimisation but it cannot be avoided.

## Linear task (optional)

https://linear.app/rudderstack/issue/SDK-4203/fix-page-unloaded-events-functionality

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [x] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

* **New Features**
  * Added an optional parameter to control cache-optimization behavior for page exit handling.

* **Improvements**
  * Page unload tracking now respects the optimization flag and has broader cross-browser handling for visibility, pagehide, focus/blur and unload sequences.
  * Expanded and reorganized test coverage for many exit scenarios and edge cases.

* **Updates**
  * Slight increase to the public bundle size limit.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: packages/analytics-js-common/src/utilities/page.ts [hunk 1]
```diff
@@ -29,7 +37,8 @@ const onPageLeave = (callback: (isAccessible: boolean) => void) => {
   // Note that 'pagehide' is not supported in IE.
   // Registering this event conditionally for IE11 also because
   // it affects bfcache optimization on modern browsers otherwise.
-  if (isIE11()) {
+  // However, if optimisation is disabled, force subscribe the event
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-common/src/utilities/page.ts]: Done.

File: packages/analytics-js-common/src/utilities/page.ts [hunk 2]
```diff
@@ -1,6 +1,14 @@
 import { isIE11 } from './detect';
 
-const onPageLeave = (callback: (isAccessible: boolean) => void) => {
+/**
+ * Registers events to detect page leave scenarios
+ * @param callback Callback function
+ * @param avoidBfCacheOptimisation When `true`, forcefully subscribes to beforeunload event, compromising on the bfcache optimisation
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-common/src/utilities/page.ts]: Done.

File: packages/analytics-js-common/src/utilities/page.ts [hunk 3]
```diff
@@ -29,7 +37,8 @@ const onPageLeave = (callback: (isAccessible: boolean) => void) => {
   // Note that 'pagehide' is not supported in IE.
   // Registering this event conditionally for IE11 also because
   // it affects bfcache optimization on modern browsers otherwise.
-  if (isIE11()) {
+  // However, if optimisation is disabled, force subscribe the event
+  if (avoidBfCacheOptimisation === true || isIE11()) {
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-common/src/utilities/page.ts]: Done.

**Review by 1abhishekpandey — APPROVED** [review]

File: packages/analytics-js-common/src/utilities/page.ts [hunk 4]
```diff
@@ -1,6 +1,14 @@
 import { isIE11 } from './detect';
 
-const onPageLeave = (callback: (isAccessible: boolean) => void) => {
+/**
+ * Registers events to detect page leave scenarios
+ * @param callback Callback function
+ * @param avoidBfCacheOptimization When `true`, forcefully subscribes to beforeunload event, compromising on the bfcache optimization
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-common/src/utilities/page.ts:6]: We can address this later.
